tools/project_manager/automodify.php

```diff
@@ -217,13 +226,17 @@ function ensure_project_blurb( $project )
         $n_hours_to_wait = 4;
         $max_reclaimable_time = time() - $n_hours_to_wait * 60 * 60;
 
-        $res = mysqli_query(DPDatabase::get_connection(), "
+        $sql = sprintf("
             SELECT image
             FROM $projectid
-            WHERE state IN ('$round->page_out_state','$round->page_temp_state')
-                AND $round->time_column_name <= $max_reclaimable_time
-            ORDER BY image ASC
-        ");
+            WHERE state IN ('%s','%s')
+                AND %d <= %d
+            ORDER BY image ASC",
+            DPDatabase::escape($round->page_out_state),
+            DPDatabase::escape($round->page_temp_state),
+            $round->time_column_name,
+            $max_reclaimable_time);
+        $res = DPDatabase::query($sql);
```

Comment:
This is a case where we don't want to raise an exception because the below handles it. It would be better to rewrite this and the next few lines in a try block:
```php
try {
    $res = DPDatabase::query($sql);
} catch(DPQueryError $error) {
    echo "$error->message\n";
    echo "Skipping further processing of this project.\n";
    continue;
}
```
The above is utterly untested and may be using slightly wrong syntax as I'm doing it from memory.